nav2_simple_commander/nav2_simple_commander/line_iterator.py

```diff
@@ -0,0 +1,141 @@
+#! /usr/bin/env python3
+# Copyright 2021 Samsung Research America
+# Copyright 2022 Afif Swaidan
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This is a Python3 API for a line iterator.
+
+It provides the ability to iterate through the points of a line.
+"""
+
+
+from cmath import sqrt
+
+
+class LineIterator():
+
+    def __init__(self, x0, y0, x1, y1):
+        """
+        Initializer for LineIterator.
+
+        Initialize instance variables with parameter occupancy_map.
+            Args:
+                x0: Abscissa of the initial point
+                y0: Ordinate of the initial point
+                x1: Abscissa of the final point
+                y1: Ordinate of the final point
+            Returns:
+                None
+        """
+        self.x0_ = x0
+        self.y0_ = x0
+        self.x1_ = x1
+        self.y1_ = y1
+        self.x_ = x0
+        self.y_ = y0
+
+        if x1 != x0 and y1 != y0:
+            self.valid_ = True
+            self.distance_ = abs(x1-x0)
+            self.m_ = (y1-y0)/(x1-x0)
+            self.b_ = y1 - (self.m_*x1)
+            if (self.b_ < 0):
+                self.equation_ = "y = " + str(self.m_) + "*x " + str(self.b_)
```

Comment:
Missing a plus or minus or something for interaction between x and b?

Reply:
In this case no since b is already negative so it will be displayed with a minus.
In the other case where b is positive I added the + sign to the string.